sagemaker-training-compiler/tensorflow/single_gpu_single_node/vision-transformer.ipynb

```diff
@@ -0,0 +1,644 @@
+{
```

Comment:
<p>Do not use .attach method. You need to use .wait function.</p><p> </p><p>See the other SMTC example notebooks: https://github.com/aws/amazon-sagemaker-examples/blob/main/sagemaker-training-compiler/huggingface/pytorch_single_gpu_single_node/albert-base-v2/albert-base-v2.ipynb</p><p> </p><pre class="ql-syntax" spellcheck="false"># Wait for training jobs to complete.

waiter_native = native_estimator.sagemaker_session.sagemaker_client.get_waiter(
    "training_job_completed_or_stopped"
)
waiter_native.wait(TrainingJobName=native_estimator.latest_training_job.name)

waiter_opt = optimized_estimator.sagemaker_session.sagemaker_client.get_waiter(
    "training_job_completed_or_stopped"
)
waiter_opt.wait(TrainingJobName=optimized_estimator.latest_training_job.name)

</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3458/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Reply:
Can you expand on why waiter is better than attach ? I tried both and found attach to be more useful as it prints more status information while waiting.

Reply:
We had the same issue during the reInvent 2021 launch that this attach method breaks for some reason conflicting with the last line in the cleaning section. If you see the build log generated as message in this PR, it says the build broke at the line with the `.attach` method. To pass CI, we have to use wait. The functionalities are the same, so it's better to use `.wait` to also pass the CI test.

Reply:
Fixed.